src/mono/wasm/runtime/managed-exports.ts

```diff
@@ -44,11 +49,29 @@ export function init_managed_exports(): void {
     mono_assert(complete_task_method, "Can't find CompleteTask method");
     const call_delegate_method = get_method("CallDelegate");
     mono_assert(call_delegate_method, "Can't find CallDelegate method");
-
-    runtimeHelpers.javaScriptExports._release_js_owned_object_by_gc_handle = (gc_handle: GCHandle) => {
-        if (!gc_handle) {
-            Module.printErr("Must be valid gc_handle");
+    runtimeHelpers.javaScriptExports.call_entry_point = async (entry_point: MonoMethod, program_args?: string[]) => {
+        const sp = anyModule.stackSave();
+        try {
+            const args = alloc_stack_frame(4);
+            const res = get_arg(args, 1);
+            const arg1 = get_arg(args, 2);
+            const arg2 = get_arg(args, 3);
+            set_arg_intptr(arg1, <any>entry_point);
+            if (program_args && program_args.length == 0) {
+                program_args = undefined;
+            }
+            marshal_array_to_cs_impl(arg2, program_args, MarshalerType.String);
+            invoke_method_and_handle_exception(call_entry_point, args);
+            const promise = marshal_task_to_js(res, undefined, marshal_int32_to_js);
+            if (!promise) {
+                return 0;
+            }
+            return await promise;
```

Comment:
i don't think stackSave/stackRestore is correct with an await involved. you need to restore the stack before yielding.

Reply:
Many thanks!